### PR TITLE
Add retry to parts of the import script which were failing from time to time

### LINKED ISSUE #243: Options search doesn't work for 20.09
At the moment, options search for 20.09 doesn't find anything. I tried different keywords that definitely exist, like "services" or "hardware" - always get the same. 

_search action returns a valid object with no errors but with zero results:

```json
{
  "took":0,
  "timed_out":false,
  "_shards":{
    "total":1,
    "successful":1,
    "skipped":0,
    "failed":0
  },
  "hits":{
    "total":{
      "value":0,
      "relation":"eq"
    },
    "max_score":null,
    "hits":[]
  }
}
```

Example URL: https://search.nixos.org/options?channel=20.09&from=0&size=30&sort=relevance&query=services

### PR DESCRIPTION
**Pull request by garbas**
fixes #243

### CODE REVIEW

**Comment by garbas** [comment]
I think this should make the import script more robust.

File: import-scripts/import_scripts/channel.py [hunk 1]
```diff
@@ -521,10 +528,11 @@ def get_options(evaluation):
     options = get_options_raw(evaluation)
 
     @functools.lru_cache(maxsize=None)
+    @backoff.on_exception(backoff.expo, subprocess.CalledProcessError)
     def jsonToNix(value):
         result = subprocess.run(
             shlex.split(
-                "nix-instantiate --eval --strict -E 'builtins.fromJSON (builtins.readFile /dev/stdin)'"
+                "nix-instantiate --eval --strict -E 'builtins.fromJSON (builtins.areadFile /dev/stdin)'"
```
Comment (ncfavier) [inline import-scripts/import_scripts/channel.py:535]: Is this a typo?

**Review by ncfavier — APPROVED** [review]
Looks good, although it would be nice to know exactly why the command is failing. I'll try and reproduce the issue locally with a standard error log.

File: import-scripts/import_scripts/channel.py (hunk 1 above)
Comment (garbas) [reply import-scripts/import_scripts/channel.py:535]: oh right, good catch. this was to test if it is failing correctly.

**Comment by garbas** [comment]
> Looks good, although it would be nice to know exactly why the command is failing. I'll try and reproduce the issue locally with a standard error log.

The problem is intermittent, I couldn't reproduce it even once locally.

**Comment by ncfavier** [comment]
> The problem is intermittent, I couldn't reproduce it even once locally.

OK

**Comment by ryantm** [comment]
I currently see no search results for 20.09 or unstable options.

**Comment by garbas** [comment]
@ryantm I hoped simply adding a retry would solve this, but it looks like we need to either revert #239 or properly implement it. And it wouldn't hurt to make the import script more robust.

**Pull request merged**